staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go

```diff
@@ -240,13 +237,19 @@ func clearSecurityContext(p *corev1.Pod, containerName string) {
 }
 
 // disallowRoot configures the container to run as a non-root user.
+// If the container is already configured to run as a non-0 UID, that will be
+// retained; otherwise the UID will be updated to 65534.
 func disallowRoot(p *corev1.Pod, containerName string) {
 	podutils.VisitContainers(&p.Spec, podutils.AllContainers, func(c *corev1.Container, _ podutils.ContainerType) bool {
 		if c.Name != containerName {
 			return true
 		}
-		c.SecurityContext = &corev1.SecurityContext{
-			RunAsNonRoot: pointer.BoolPtr(true),
+		if c.SecurityContext == nil {
+			c.SecurityContext = &corev1.SecurityContext{}
+		}
+		c.SecurityContext.RunAsNonRoot = pointer.BoolPtr(true)
+		if c.SecurityContext.RunAsUser == nil || *c.SecurityContext.RunAsUser == 0 {
+			c.SecurityContext.RunAsUser = pointer.Int64(65534)
```

Comment:
Why we are using this special `65534`?. I think, we need to set something different other than some static value.

Reply:
verb just responded to the issue and looks like the `restricted` profile should be used where the debug container image itself specifies a non-root user, e.g.[ this container image](https://github.com/verb/dockerfiles/blob/master/busybox-nonroot/Dockerfile).

So I've just pushed an update to no longer explicitly set a runAsUser and pulled back the scope of this MR to the other fixes which is that we now set runAsNonRoot and drop-all capabilities without having side-effects on other unrelated sescurity context properties.